File: data/src/main/java/com/github/chojmi/inspirations/data/entity/photos/PhotoFavsEntityImpl.java
package com.github.chojmi.inspirations.data.entity.photos;

import android.os.Parcelable;

import com.github.chojmi.inspirations.domain.entity.photos.PhotoFavsEntity;
import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

import java.util.List;

@AutoValue
public abstract class PhotoFavsEntityImpl implements Parcelable, PhotoFavsEntity {
    public static TypeAdapter<PhotoFavsEntityImpl> typeAdapter(Gson gson) {
        return new AutoValue_PhotoFavsEntityImpl.GsonTypeAdapter(gson);
    }

    @SerializedName("person")
    public abstract List<PersonEntityImpl> getPeople();

    public abstract int getPage();

    @SerializedName("id")
    public abstract String getPhotoId();

    public abstract int getTotal();
}

File: data/src/main/java/com/github/chojmi/inspirations/data/entity/photos/PhotoFavsEntityImpl.kt
package com.github.chojmi.inspirations.data.entity.photos

import com.github.chojmi.inspirations.domain.entity.photos.PhotoFavsEntity
import com.google.gson.annotations.SerializedName

data class PhotoFavsEntityImpl(private @SerializedName("person") val people: List<PersonEntityImpl>,
                               private val page: Int,
                               private @SerializedName("id") val photoId: String,
                               private val total: Int) : PhotoFavsEntity<PersonEntityImpl> {
    override fun getPeople(): List<PersonEntityImpl> = people

    override fun getPage(): Int = page

    override fun getPhotoId(): String = photoId

    override fun getTotal(): Int = total
}